### PR TITLE
fix(Test Coverage): Improve test coverage of 2 core files

### LINKED ISSUE #690: fix(Test Coverage): Improve test coverage of 2 core files
These files are well below the coverage of other files in core and need to be brought up to at least 80%:

kinesisanalytics-helper.ts (38%)
sqs-helper.ts (71%)

### PR DESCRIPTION
**Pull request by biffgaut**
*Issue #, if available:*
closes #690 

*Description of changes:*
Added additional tests for both modules to improve line coverage to >90% 
By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

**Comment by aws-solutions-constructs-team** [comment]
### AWS CodeBuild CI Report

* CodeBuild project: githubautobuild-for-cdk-v2
* Commit ID: a87231c289dbfa69179c91b67e3a01f97793fbb1
* Result: SUCCEEDED
* [Build Logs](https://j8dgcooz5f.execute-api.us-east-1.amazonaws.com/Prod/buildlogs?key=b9ee04e1-ded6-4f93-af80-12d088a70bde%2Fbuild.log) (available for 30 days)

*Powered by [github-codebuild-logs](https://serverlessrepo.aws.amazon.com/applications/arn:aws:serverlessrepo:us-east-1:277187709615:applications~github-codebuild-logs), available on the [AWS Serverless Application Repository](https://aws.amazon.com/serverless/serverlessrepo/)*

**Comment by aws-solutions-constructs-team** [comment]
### AWS CodeBuild CI Report

* CodeBuild project: githubautobuild-for-cdk-v1
* Commit ID: a87231c289dbfa69179c91b67e3a01f97793fbb1
* Result: SUCCEEDED
* [Build Logs](https://nis3u8ftb6.execute-api.us-east-1.amazonaws.com/Prod/buildlogs?key=143a1576-88d9-4bd9-8a9b-a82c0f1fa02d%2Fbuild.log) (available for 30 days)

*Powered by [github-codebuild-logs](https://serverlessrepo.aws.amazon.com/applications/arn:aws:serverlessrepo:us-east-1:277187709615:applications~github-codebuild-logs), available on the [AWS Serverless Application Repository](https://aws.amazon.com/serverless/serverlessrepo/)*

**Comment by aws-solutions-constructs-team** [comment]
### AWS CodeBuild CI Report

* CodeBuild project: githubautobuild-for-cdk-v2
* Commit ID: ef42ea56196110b3376d4a6135a2564697e0ae83
* Result: SUCCEEDED
* [Build Logs](https://j8dgcooz5f.execute-api.us-east-1.amazonaws.com/Prod/buildlogs?key=0e7c694a-344b-4479-89c9-aa000f99dd4c%2Fbuild.log) (available for 30 days)

*Powered by [github-codebuild-logs](https://serverlessrepo.aws.amazon.com/applications/arn:aws:serverlessrepo:us-east-1:277187709615:applications~github-codebuild-logs), available on the [AWS Serverless Application Repository](https://aws.amazon.com/serverless/serverlessrepo/)*

**Comment by aws-solutions-constructs-team** [comment]
### AWS CodeBuild CI Report

* CodeBuild project: githubautobuild-for-cdk-v1
* Commit ID: ef42ea56196110b3376d4a6135a2564697e0ae83
* Result: SUCCEEDED
* [Build Logs](https://nis3u8ftb6.execute-api.us-east-1.amazonaws.com/Prod/buildlogs?key=089e0c19-a7ac-4ad2-99d3-75dd22802936%2Fbuild.log) (available for 30 days)

*Powered by [github-codebuild-logs](https://serverlessrepo.aws.amazon.com/applications/arn:aws:serverlessrepo:us-east-1:277187709615:applications~github-codebuild-logs), available on the [AWS Serverless Application Repository](https://aws.amazon.com/serverless/serverlessrepo/)*

File: source/patterns/@aws-solutions-constructs/core/test/sqs-helper.test.ts [hunk 1]
```diff
@@ -57,3 +59,93 @@ test('Test deployment without imported encryption key', () => {
...
+  const stack = new Stack();
+
+  const existingQueue = new sqs.Queue(stack, 'test-queue');
+  const buildDlqProps: defaults.BuildDeadLetterQueueProps = {
+    existingQueueObj: existingQueue,
+  };
+
+  const returnedQueueu = defaults.buildDeadLetterQueue(stack, buildDlqProps);
+
+  expect(returnedQueueu).toBeUndefined();
```
Comment (mickychetta) [inline source/patterns/@aws-solutions-constructs/core/test/sqs-helper.test.ts:97]: Minor, queue typo

**Review by mickychetta — CHANGES_REQUESTED** [review]
Confusing behavior around deadLetterQueue prop, add more functional tests for DLQ to confirm proper creation/instantiation of queues:
```
    // Determine whether a DLQ property should be added
    if (props.deadLetterQueue) {
      queueProps.deadLetterQueue = props.deadLetterQueue;
    }
```

**Comment by aws-solutions-constructs-team** [comment]
### AWS CodeBuild CI Report

* CodeBuild project: githubautobuild-for-cdk-v2
* Commit ID: 8071e0a4fd41abc759e581419157d6b2074fff25
* Result: SUCCEEDED
* [Build Logs](https://j8dgcooz5f.execute-api.us-east-1.amazonaws.com/Prod/buildlogs?key=c72d3fd8-3c0e-4a08-adb3-25a3c31e13e4%2Fbuild.log) (available for 30 days)

*Powered by [github-codebuild-logs](https://serverlessrepo.aws.amazon.com/applications/arn:aws:serverlessrepo:us-east-1:277187709615:applications~github-codebuild-logs), available on the [AWS Serverless Application Repository](https://aws.amazon.com/serverless/serverlessrepo/)*

**Comment by aws-solutions-constructs-team** [comment]
### AWS CodeBuild CI Report

* CodeBuild project: githubautobuild-for-cdk-v1
* Commit ID: 8071e0a4fd41abc759e581419157d6b2074fff25
* Result: SUCCEEDED
* [Build Logs](https://nis3u8ftb6.execute-api.us-east-1.amazonaws.com/Prod/buildlogs?key=012f2fb2-1bfb-48b1-9837-21fc8430da0f%2Fbuild.log) (available for 30 days)

*Powered by [github-codebuild-logs](https://serverlessrepo.aws.amazon.com/applications/arn:aws:serverlessrepo:us-east-1:277187709615:applications~github-codebuild-logs), available on the [AWS Serverless Application Repository](https://aws.amazon.com/serverless/serverlessrepo/)*

**Review by mickychetta — APPROVED** [review]

**Comment by aws-solutions-constructs-team** [comment]
### AWS CodeBuild CI Report

* CodeBuild project: githubautobuild-for-cdk-v1
* Commit ID: b02489c64b0792be3e69b91c272e3424f14551df
* Result: SUCCEEDED
* [Build Logs](https://nis3u8ftb6.execute-api.us-east-1.amazonaws.com/Prod/buildlogs?key=7cb0abcb-3d49-42ef-98cb-0b6b5b70480a%2Fbuild.log) (available for 30 days)

*Powered by [github-codebuild-logs](https://serverlessrepo.aws.amazon.com/applications/arn:aws:serverlessrepo:us-east-1:277187709615:applications~github-codebuild-logs), available on the [AWS Serverless Application Repository](https://aws.amazon.com/serverless/serverlessrepo/)*

**Pull request merged**